src/main/java/org/lightcouch/View.java

```diff
@@ -583,7 +603,22 @@ public View tempView(MapReduce mapReduce) {
     }
 
     private String getKeyAsJson(Object... key) {
-        return (key.length == 1) ? gson.toJson(key[0]) : gson.toJson(key); // single or complex key
```

Comment:
Yes, the two methods are similar.  I believe it shouldn't be difficult to make the required changes in the ```key(Object... key)``` and ```keys(Object... keys)``` to call ```getKeyAsJsonElement()```.